aggregator/src/aggregator.rs

```diff
@@ -1892,8 +1899,35 @@ impl VdafOps {
                         .add(1, &[KeyValue::new("type", "duplicate_extension")]);
                     return Err(PrepareError::InvalidMessage);
                 }
+
+                if require_taskprov_extension {
+                    if !extensions.get(&ExtensionType::Taskprov).map(|data| data.is_empty()).unwrap_or(false) {
+                        debug!(
+                            task_id = %task.id(),
+                            metadata = ?prepare_init.report_share().metadata(),
+                            "Taskprov task received report with missing or malformed taskprov extension",
+                        );
+                        aggregate_step_failure_counter
+                            .add(1, &[KeyValue::new("type", "missing_or_malformed_taskprov_extension")]);
+                        taskprov_error = Some(Error::InvalidMessage(Some(*task.id()), "missing or malformed taskprov extension"));
```

Comment:
The layout of this logic means that, if we receive a valid taskprov header in an aggregation initialization where some ReportShares don't contain the new taskprov extension, we will abort the aggregation but leave the task in place.

I think that's OK -- tasks can still only be created by authorized parties, and even if the task is created in error, the cost of an unused task sitting around is minimal.

More practically, it would be challenging to do otherwise: either we would need to check the extensions before we created the task (which would imply parsing the request, which is query-type specific, and decrypting the plaintext input shares), or we would need to unwind the task creation on failure (and think through possible race conditions with other aggregations).